src/dune_patch/dune_patch.ml

```diff
@@ -44,25 +42,22 @@ let patches_of_string patch_string =
   Re.all re patch_string
   |> List.filter_map ~f:(fun group ->
     let open Option.O in
-    (* If these are [Some] then they are [/dev/null] *)
-    let old_file_dev_null = Re.Group.get_opt group 2 in
-    let new_file_dev_null = Re.Group.get_opt group 4 in
-    match old_file_dev_null, new_file_dev_null with
-    | Some _dev_null, Some _ ->
+    (* A match failure means a file name couldn't be parsed. *)
+    let* old_file = Re.Group.get_opt group 1 in
+    let* new_file = Re.Group.get_opt group 2 in
+    match old_file = "/dev/null", new_file = "/dev/null" with
```

Comment:
Is this a change relevant to the bug fix?

Reply:
They're part of fixing this bug and part of fixing a related bug found while reading the old code: any file not starting with "a/" is interpreted as "/dev/null"

I can make a repro PR if you want

Reply:
If the file doesn't start with some prefix, be it `a/`/`b/` or some other prefix, we can't apply it anyway since we pass `-p1` to `patch` and `patch` will not apply it.

Thus I think the right thing would be to make sure that there is a prefix (or the file is `/dev/null`) or call `patch` with `-p0` if there was no prefix.